src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -97,19 +97,36 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
+			//Gets all modules of the type specified by variable type
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
 		if (query != null) {
-			boolQB.should(QueryBuilders.matchQuery("name", query).boost(4.0f));
-			boolQB.should(QueryBuilders.prefixQuery("_all", query).boost(1.5f));
-			boolQB.should(QueryBuilders.matchPhrasePrefixQuery("_all", query).slop(2).fuzziness("AUTO"));
+			//Following clause returns a module whose UID matches exactly(Highest priority)
```

Comment:
Like I said in the comment above this one, you should make the comment shorter and more efficient.

* Don't say "Following clause" because it's obvious from the code spacing that this comment applies to the next line
* Don't say "returns all modules" because anyone reading this block of code knows that we're building a query, and hence it's going to query for a group of addons.

I would have written this as

    // Top priority: exact match on _id

(And, apply this same comment to the next 4 comments also.